numpy/f2py/__init__.py

```diff
@@ -62,7 +63,26 @@ def compile(source,
         args = ' -c -m {} {} {}'.format(modulename, f.name, extra_args)
         c = '{} -c "import numpy.f2py as f2py2e;f2py2e.main()" {}'
         c = c.format(sys.executable, args)
-        status, output = exec_command(c)
+        # historically, exec_command() would carefully curate
+        # list-based command input into a string, with separate
+        # Windows and POSIX handling scenarios; here, the c object
+        # is already a string, but we are using subprocess, which
+        # wants a list of strings for commands; we will split on
+        # whitespace to perform the opposite conversion -- presumably
+        # this is portable
+        c = c.split(' ')
```

Comment:
review of related PRs has generally been -1 for splitting on whitespace -- here, the conventional build / tests are not sensitive to the presence of this line, but I am concerned that `c` is clearly a `str` here & that there are double quotes around the numpy import line.

Reply:
Yeah, this will not work for sure.You want:

```
c = [sys.executable, '-c', 'import numpy.f2py as f2py2e;f2py2e.main()'] + args
```

Where `args` is defined similarly